From: Hendrik Brueckner <brueckner@redhat.com>
Date: Fri, 4 Dec 2009 13:57:42 -0500
Subject: [s390] zcrypt: adjust speed rating of cex3 adapters
Message-id: <20091204135953.132801294@redhat.com>
Patchwork-id: 21686
O-Subject: [RHEL6 PATCH 6/6] s390 - zcrypt: adjust speed rating of cex3 adapters
Bugzilla: 537495
RH-Acked-by: Pete Zaitcev <zaitcev@redhat.com>

Description
===========
Cex3 needs a lower speed rating. Otherwise cex2 adapters will be
prefered.

Bugzilla
========
BZ 537495
https://bugzilla.redhat.com/show_bug.cgi?id=537495

Upstream status of the patch
============================
The patch will be upstream as of kernel version 2.6.33.
The patch has been already posted to LKML:
http://marc.info/?l=linux-kernel&m=125812564316752&w=4

Test status
===========
The patch has been tested and fixes the problem.
The fix has been verified by the IBM test department.

Please ACK.

With best regards,

	Hendrik

Signed-off-by: Aristeu Rozanski <arozansk@redhat.com>

diff --git a/drivers/s390/crypto/zcrypt_cex2a.c b/drivers/s390/crypto/zcrypt_cex2a.c
index a90b760..c6fb0aa 100644
--- a/drivers/s390/crypto/zcrypt_cex2a.c
+++ b/drivers/s390/crypto/zcrypt_cex2a.c
@@ -43,7 +43,7 @@
 #define CEX3A_MAX_MOD_SIZE	CEX2A_MAX_MOD_SIZE
 
 #define CEX2A_SPEED_RATING	970
-#define CEX3A_SPEED_RATING	1100
+#define CEX3A_SPEED_RATING	900 /* Fixme: Needs finetuning */
 
 #define CEX2A_MAX_MESSAGE_SIZE	0x390	/* sizeof(struct type50_crb2_msg)    */
 #define CEX2A_MAX_RESPONSE_SIZE 0x110	/* max outputdatalength + type80_hdr */
diff --git a/drivers/s390/crypto/zcrypt_pcixcc.c b/drivers/s390/crypto/zcrypt_pcixcc.c
index 65cf17b..79c1205 100644
--- a/drivers/s390/crypto/zcrypt_pcixcc.c
+++ b/drivers/s390/crypto/zcrypt_pcixcc.c
@@ -49,7 +49,7 @@
 #define PCIXCC_MCL2_SPEED_RATING	7870
 #define PCIXCC_MCL3_SPEED_RATING	7870
 #define CEX2C_SPEED_RATING		7000
-#define CEX3C_SPEED_RATING		10000	/* FIXME: needs finetuning */
+#define CEX3C_SPEED_RATING		6500	/* FIXME: needs finetuning */
 
 #define PCIXCC_MAX_ICA_MESSAGE_SIZE 0x77c  /* max size type6 v2 crt message */
 #define PCIXCC_MAX_ICA_RESPONSE_SIZE 0x77c /* max size type86 v2 reply	    */
